pyhealth/datasets/base_ehr_dataset.py

```diff
@@ -135,6 +135,25 @@ def __init__(
             logger.debug(f"Saved {self.dataset_name} base dataset to {self.filepath}")
             save_pickle(self.patients, self.filepath)
 
+        self.patient_ids = list(self.patients.keys()) # precompute for use in __getitem__
+
+    def __len__(self):
+        return len(self.patients)
+
+    """
+    access dataset like dataset[10], or convert dataset to list via list(dataset)
+    """
+    def __getitem__(self, index: slice):
+        if (isinstance(index, slice)):
+            patient_ids = self.patient_ids[index]
+            res = []
+            for p in patient_ids:
+                res.append (self.patients[p])
+            return res
+        else:
+            patient_id = self.patient_ids[index]
+            return self.patients[patient_id]
+
```

Comment:
the above allows us to slice the dataset by patients. I use this to make train/test/val splits without transforming data into a SampleEHRDataset (which doesn't fit my application)